.github/workflows/ruff.yml

```diff
@@ -8,6 +8,5 @@ jobs:
     steps:
     - uses: actions/checkout@v4
     # ruff version should be kept in sync with .pre-commit-config.yaml
-    - run: pip install --user ruff==0.1.3
-    - run: ruff check --output-format=github .
-    - run: ruff format --diff .
+    - run: pipx run ruff==0.9.6 check --output-format=github .
```

Comment:
Why not keep the existing `pip install` and just change the version to 0.9.6?  At least then you only need to specify the version once.